service/storage_proxy.cc

```diff
@@ -2989,18 +2997,22 @@ class digest_read_resolver : public abstract_read_resolver {
             _done_promise.set_value(bo::success());
         }
     }
-    void on_error(gms::inet_address ep, bool disconnect) override {
+    void on_error(gms::inet_address ep, error_kind kind) override {
         if (waiting_for(ep)) {
             _failed++;
         }
-        if (disconnect && _block_for == _target_count_for_cl) {
+        if (kind == error_kind::DISCONNECT && _block_for == _target_count_for_cl) {
             // if the error is because of a connection disconnect and there is no targets to speculate
             // wait for timeout in hope that the client will issue speculative read
             // FIXME: resolver should have access to all replicas and try another one in this case
             return;
         }
         if (_block_for + _failed > _target_count_for_cl) {
-            fail_request(read_failure_exception(_schema->ks_name(), _schema->cf_name(), _cl, _cl_responses, _failed, _block_for, _data_result));
+            if (kind == error_kind::RATE_LIMIT) {
```

Comment:
It's better to use `switch` (and no `default`) for enums so adding an enum member is safer as the compiler forces us to check call the sites.

Reply:
Done